src/react-components/auth/AuthContext.js

```diff
@@ -0,0 +1,96 @@
+import React, { createContext, useState, useEffect, useCallback } from "react";
+import PropTypes from "prop-types";
+import configs from "../../utils/configs";
+
+// TODO: We really shouldn't include these dependencies on every page. A dynamic import would work better.
+import jwtDecode from "jwt-decode";
+import AuthChannel from "../../utils/auth-channel";
+import { connectToReticulum } from "../../utils/phoenix-utils";
+
+export const AuthContext = createContext();
+
+export function AuthContextProvider({ children, store }) {
+  const signIn = useCallback(
+    async email => {
+      const authChannel = new AuthChannel(store);
+      const socket = await connectToReticulum();
+      authChannel.setSocket(socket);
+      const { authComplete } = await authChannel.startAuthentication(email);
+      await authComplete;
+
+      // TODO: Doing all of this just to determine if the user is an admin seems unnecessary. The auth callback should have the isAdmin flag.
+      const retPhxChannel = socket.channel("ret", { hub_id: "index", token: store.state.credentials.token });
+
+      const perms = await new Promise(resolve => {
+        retPhxChannel.join().receive("ok", () => {
+          retPhxChannel.push("refresh_perms_token").receive("ok", ({ perms_token }) => {
+            const perms = jwtDecode(perms_token);
+            retPhxChannel.leave();
+            resolve(perms);
+          });
+        });
+      });
+
+      configs.setIsAdmin(perms.postgrest_role === "ret_admin");
+      store.update({ credentials: { isAdmin: perms.postgrest_role === "ret_admin" } });
+    },
+    [store]
```

Comment:
Just to confirm my understanding, we don't actually ever expect store to change correct?

Reply:
No, but the rules of hooks expect it to be passed in as a dependency. eslint will complain otherwise.